homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -248,54 +205,73 @@ def _async_register_button_devices(
 
 
 @callback
-def _async_subscribe_pico_remote_events(hass, lip, button_devices_by_id):
+def _async_subscribe_pico_remote_events(
+    hass: HomeAssistant,
+    bridge_device: Smartbridge,
+    button_devices_by_id: dict[int, dict],
+):
     """Subscribe to lutron events."""
 
     @callback
-    def _async_lip_event(lip_message):
-        if lip_message.mode != LIPMode.DEVICE:
-            return
-
-        device = button_devices_by_id.get(lip_message.integration_id)
+    def _async_button_event(button_id, event_type):
+        device = button_devices_by_id.get(button_id)
 
         if not device:
             return
 
-        if lip_message.value == LIP_BUTTON_PRESS:
+        if event_type == BUTTON_STATUS_PRESSED:
             action = ACTION_PRESS
         else:
             action = ACTION_RELEASE
 
+        type_ = device["type"]
+        name = device["name"]
+        button_number = device["button_number"]
+        # The original implementation used LIP instead of LEAP
+        # so we need to convert the button number to maintain compat
+        sub_type_to_lip_button = DEVICE_TYPE_SUBTYPE_MAP_TO_LIP[type_]
+        leap_button_to_sub_type = LEAP_TO_DEVICE_TYPE_SUBTYPE_MAP[type_]
+        if (sub_type := leap_button_to_sub_type.get(button_number)) is None:
+            _LOGGER.error(
+                "Unknown LEAP button number %s is not in %s for %s (%s)",
+                button_number,
+                leap_button_to_sub_type,
+                name,
+                type_,
+            )
+            return
+        lip_button_number = sub_type_to_lip_button[sub_type]
+
         hass.bus.async_fire(
             LUTRON_CASETA_BUTTON_EVENT,
             {
-                ATTR_SERIAL: device.get("serial"),
-                ATTR_TYPE: device.get("type"),
-                ATTR_BUTTON_NUMBER: lip_message.action_number,
-                ATTR_DEVICE_NAME: device["Name"],
-                ATTR_AREA_NAME: device.get("Area", {}).get("Name"),
+                ATTR_SERIAL: device["serial"],
+                ATTR_TYPE: type_,
+                ATTR_BUTTON_NUMBER: lip_button_number,
```

Comment:
Would it make sense to also include the LEAP button number?

Reply:
Someone will probably ask 👍

Reply:
done